releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+
+Changes to the current e2e tests would be minimal initially.
+- Testing of cluster-$provider app would require one addition, which would be:
+	- Looking up the latest release and setting the release label on the App CR.
+	- Like today, cluster-test-suites would use custom version of cluster-$provider app from the branch it is testing (in which case the mutating webhook would not set the cluster-$provider app version based on the Release CR).
+- The existing app testing where app versions are overridden via ephemeral Helm values would continue working as is.
+- As a future improvement, cluster-test-suites could run all tests not only for the latest release, but for all releases which use the same major cluster-$provider app version.
+
+#### 4.2.4. Multiple release models and release channels
+
+With cluster-$provider app releases, while we can use git branches to have multiple major versions, all that is only one way to create releases. We can have multiple major versions, but it’s still the same apps being versioned in the same way in every release.
+
+Since app and component versions are embedded in the cluster-$provider app, we cannot combine different versions of different apps in different ways (unless we make app and component versions configurable, which then opens the door to a whole new set of issues). We cannot have another release model where e.g. versions of Kubernetes, OS, CNI and CPI are a part of the release, and all other apps are always at their latest versions. Or a release model where we use LTS release of OS, CPI, CNI and other apps that provide a LTS release (or something similar).
+
+OTOH with the releases repository, we can easily create different directories for different release models, where we can combine the versions of apps and components in whatever way and where we can update different release models with different frequency and with different rules. Therefore it would be relatively easy to have one release model where app versions are not even part of the release, or are decoupled and continuously updated. Or another “slower” release model with long-term support where apps and components are on their LTS versions and are updated more slowly.
+
+## 5. Conclusion
+
+They current cluster-$provider releases are attractive thanks to their simplicity and straightforwardness. However they have not been put to a test yet, as our customers are still not using the new product for their production workloads.
+
+As soon as our customers move from our vintage product to Cluster API, we will have a need for multiple major releases. Once the customers are using the new product more, they will put it to test and we’ll have even more work on it, compared to today, so there will be even more need for more patch and minor releases across all providers.
```

Comment:
This leaves me thinking if we have a plan and timeline how quick we can move to this model, as we already have a timeline when customers will be moving and we also need a plan how to move customers that already are using the existing system. cc @alex-dabija

Reply:
What about right away? 🙈 😬 

Required changes are very small, I have a working POC already, with changes in cluster chart, cluster-aws and app-admission-controller.

I have added an agenda item for this week's SIG Architecture to show it there.

Reply:
We will definitely need to also look at how upgrades are handled, but generally good if we can move fast.

Reply:
Upgrade of existing clusters would be just two changes (slightly different compared to a regular version bump):
- Remove spec.version in cluster-$provider app manifest.
- Set release version.

Then for clusters on the new release system (with Release CRs) it's just a regular version bumps like for cluster-aws version.